src/main/resources/Diagram/MigrationScript.xml

```diff
@@ -0,0 +1,157 @@
+<?xml version="1.0" encoding="UTF-8"?>
+
+<!--
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+-->
+
+<xwikidoc version="1.3" reference="Diagram.MigrationScript" locale="">
+  <web>Diagram</web>
+  <name>MigrationScript</name>
+  <language/>
+  <defaultLanguage/>
+  <translation>0</translation>
+  <creator>xwiki:XWiki.Admin</creator>
+  <creationDate>1549541592000</creationDate>
+  <parent>xwiki:Diagram.WebHome</parent>
+  <author>xwiki:XWiki.Admin</author>
+  <contentAuthor>xwiki:XWiki.Admin</contentAuthor>
+  <date>1549541642000</date>
+  <contentUpdateDate>1549541592000</contentUpdateDate>
+  <version>1.1</version>
+  <title>MigrationScript</title>
+  <comment/>
+  <minorEdit>false</minorEdit>
+  <syntaxId>xwiki/2.1</syntaxId>
+  <hidden>true</hidden>
+  <content>{{velocity}}
+
+#macro(selectDiagramsRequiringMigration)
+  #set ($repository = $services.extension.getRepository('installed'))
+  #set ($extension = $services.extension.installed.getInstalledExtension('org.xwiki.contrib:draw.io', "wiki:$xcontext.database"))
+  #if ($extension != $NULL)
+    #set ($targetVersion = $extension.version)
+  #end
+  #if ($targetVersion)
+    #set ($migrationIsRequiredClause = " and doc.content like '%draw.io%' and doc.content not like '%draw.io/$targetVersion/%'")
+    #set ($hql = "select doc.fullName from XWikiDocument as doc, BaseObject as obj where obj.name = doc.fullName and obj.className='Diagram.DiagramClass' and doc.fullName &lt;&gt; 'Diagram.DiagramTemplate' $migrationIsRequiredClause")
+    #set ($toBeMigratedEntries = $services.query.hql($hql).execute())
+  #end
+#end
+
+## Display the content below only if the page is not included by another one.
+#if ($doc.name == 'MigrationScript')
```

Comment:
You don't need this hack if you create a separate page with Velocity macros and include it in the Diagram home page.

Reply:
Right, I removed it indeed.